### PR TITLE
Align sources with upstream

### PR DESCRIPTION
**Pull request by sprt**
Related PR: https://github.com/microsoft/CBL-Mariner/pull/6942

This cherry picks our `cc-msft-prototypes` changes on top of upstream main. Vanilla sources are out of scope for this PR.

Conformance run on 8aec434: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=470892

### CODE REVIEW

File: src/runtime/Makefile
Comment (sprt) [inline src/runtime/Makefile:275]: Potentially default this to false. In any case, we'll have to define this variable when building either vanilla/CC.

File: src/runtime/config/configuration-clh.toml.in
Comment (sprt) [inline src/runtime/config/configuration-clh.toml.in:43]: Will need to make this a variable for vanilla.

Comment (danmihai1) [reply src/runtime/config/configuration-clh.toml.in:43]: In this case I am tempted to create a new *.in file, instead of modifying configuration-clh.toml.in.

That's to be consistent with the upstream convention - they have these separate files:

configuration-qemu.toml.in
configuration-qemu-tdx.toml.in

It seems like our file should be named:

configuration-clh-snp.toml.in

File: src/agent/samples/policy/outdated/ctr/ctr-busybox-data.json
Comment (danmihai1) [inline src/agent/samples/policy/outdated/ctr/ctr-busybox-data.json:1]: If you want to reduce the diff, you can delete all the files under outdated/. Those files haven't been functional for a several months now.

File: src/agent/samples/policy/set-policy-allowed/set-policy-allowed-data.json
Comment (danmihai1) [inline src/agent/samples/policy/set-policy-allowed/set-policy-allowed-data.json:1]: This file will not be used anymore when/if we'll adopt the rootfs.sh changes for Policy + OPA.

File: src/runtime/config/configuration-clh.toml.in
Comment (danmihai1) [inline src/runtime/config/configuration-clh.toml.in:234]: Everyone, let's try to avoid unnecessary changes in the future.

File: tools/osbuilder/rootfs-builder/cbl-mariner/rootfs_lib.sh
Comment (danmihai1) [inline tools/osbuilder/rootfs-builder/cbl-mariner/rootfs_lib.sh:26]: If I remember correctly, Mitch was working on getting rid of REMOVED_PACKAGES.

File: src/runtime/config/configuration-clh.toml.in
Comment (sprt) [reply src/runtime/config/configuration-clh.toml.in:43]: Yep we do have a configuration-clh-snp.toml.in file in this change.

File: tools/osbuilder/rootfs-builder/cbl-mariner/rootfs_lib.sh
Comment (sprt) [reply tools/osbuilder/rootfs-builder/cbl-mariner/rootfs_lib.sh:26]: Right, we'll have to wait until his changes are merged to make the changes here.

**Comment by ms-mahuber** [comment]
General comment. To assess which of the patch files from 'https://github.com/microsoft/CBL-Mariner/tree/2.0/SPECS/kata-containers' (resp. remaining files in https://github.com/microsoft/CBL-Mariner/tree/mfrw/kata-containers-3.2.0/SPECS/kata-containers) are still missing here. 

None of our SPECS should live with those patch files, if possible.

One known missing one is: [runtime-reduce-uvm-high-mem-footprint.patch](https://github.com/microsoft/CBL-Mariner/blob/mfrw/kata-containers-3.2.0/SPECS/kata-containers/runtime-reduce-uvm-high-mem-footprint.patch). To assess.

File: src/runtime/Makefile
Comment (sprt) [reply src/runtime/Makefile:275]: This is only present in CCv0, not main, hence why it's true by default. For vanilla, we'll want to introduce a variable and set this when building in the spec.

File: src/runtime/config/configuration-clh.toml.in
Comment (sprt) [reply src/runtime/config/configuration-clh.toml.in:43]: Out of those two, I vote to create the third file dynamically in the spec since it's a trivial operation.

But also, since this is only used for debugging, I'm thinking we shouldn't package that third config at all. It would add another step to the debugging process, but this could be done as part of a broader debug script and could be an opportunity to streamline our dev process. Food for thought.

**Comment by sprt** [comment]
@ms-mahuber Good call out!

From CBL-Mariner 2.0:

* 0001-Append-systemd-kernel-cmdline-params-for-initrd.patch: https://github.com/microsoft/kata-containers/commit/2ec9760a246ab1fbadafa7ce301a02116a5e870d
* 0001-Merged-PR-9607-Allow-10-seconds-for-VM-creation-star.patch: https://github.com/microsoft/kata-containers/commit/0e6662712f0cb34fadc8f439e90ef35c76849fd1
* 0001-osbuilder-Add-support-for-CBL-Mariner.patch: upstreamed a while ago and in this branch.
* 0002-Merged-PR-9671-Wait-for-a-possibly-slow-Guest.patch: dial_timeout currently set to the upstream default of 45 in this branch, TBD if we want to keep 60.
* 0003-Merged-PR-9805-Add-support-for-MSHV.patch: upstreamed a while ago
* 0004-Merged-PR-9806-Fix-enable_debug-for-hypervisor.clh.patch: ❌ missing from this branch
* 0005-Merged-PR-9956-shim-avoid-memory-hotplug-timeout.patch: ❌ missing from this branch
* drop-mut-for-variables-that-are-not-mutated.patch: done upstream in https://github.com/kata-containers/kata-containers/commit/b9f100b3918822405277bfb694b1c8636c344714
* runtime-reduce-uvm-high-mem-footprint.patch: ❌ missing in this branch

Note the patches in Falak's branch are a subset of the above.

File: src/runtime/Makefile
Comment (ms-mahuber) [reply src/runtime/Makefile:275]: Followed up offline here, looks good, our forked code seems to need this.

File: src/agent/samples/policy/all-allowed/all-allowed.rego
Comment (sprt) [inline src/agent/samples/policy/all-allowed/all-allowed.rego:1]: This file is in src/kata-opa/ now so can probably be removed. @danmihai1?

**Comment by ms-mahuber** [comment]
> @ms-mahuber Good call out!
> 
> From CBL-Mariner 2.0:
> 
> * 0001-Append-systemd-kernel-cmdline-params-for-initrd.patch: [2ec9760](https://github.com/microsoft/kata-containers/commit/2ec9760a246ab1fbadafa7ce301a02116a5e870d)
> * 0001-Merged-PR-9607-Allow-10-seconds-for-VM-creation-star.patch: [0e66627](https://github.com/microsoft/kata-containers/commit/0e6662712f0cb34fadc8f439e90ef35c76849fd1)
> * 0001-osbuilder-Add-support-for-CBL-Mariner.patch: upstreamed a while ago and in this branch
> * 0002-Merged-PR-9671-Wait-for-a-possibly-slow-Guest.patch: 🟡 dial_timeout currently set to the upstream default of 45 in this branch, TBD if we want to keep 60 as in cc-msft-prototypes
> * 0003-Merged-PR-9805-Add-support-for-MSHV.patch: upstreamed a while ago
> * 0004-Merged-PR-9806-Fix-enable_debug-for-hypervisor.clh.patch: ❌ missing from this branch
> * 0005-Merged-PR-9956-shim-avoid-memory-hotplug-timeout.patch: ❌ missing from this branch
> * drop-mut-for-variables-that-are-not-mutated.patch: done upstream in [kata-containers@b9f100b](https://github.com/kata-containers/kata-containers/commit/b9f100b3918822405277bfb694b1c8636c344714)
> * runtime-reduce-uvm-high-mem-footprint.patch: ❌ missing in this branch
> 
> Note the patches in Falak's branch are a subset of the above.

After discussion, to include runtime-reduce-uvm-high-mem-footprint.patch

**Comment by sprt** [comment]
Only going to include the memory patch as discussed offline.

**Review by danmihai1 — APPROVED** [review]

**Pull request merged**
